.github/workflows/ci_ubuntu.yml

```
@@ -0,0 +1,72 @@
+name: bandersnatch_ci_ubuntu
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: bandersnatch CI python ${{ matrix.python-version }} on ${{matrix.os}}
+    # We want to run on external PRs, but not on our own internal PRs (for the
+    # pull_request event) as they'll be run by the push to the branch. Without
+    # this if check, ci workflow checks are duplicated since internal PRs match
+    # both the push and pull_request events.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: ${{ matrix.os }}
+
+    services:
+      minio:
+        image: lazybit/minio:latest
+        ports:
+          - 9000:9000
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9]
+        os: [ubuntu-latest]
+
+    steps:
+    - uses: actions/checkout@v2.3.4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2.2.2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install latest pip, setuptools + tox
+      run: |
+        python -m pip install --upgrade pip setuptools tox
+
+    - name: Install base bandersnatch requirements
+      run: |
+        python -m pip install -r requirements.txt
+
+    - name: Run Unittests
+      env:
+        TOXENV: py3
+        os: ${{ matrix.os }}
+      run: |
+        python test_runner.py
+
+    - name: Upload coverage to Codecov
+      # If pushed / is a pull request against main repo.
+      if: (github.event_name == 'push' && github.repository == 'pypa/bandersnatch') || (github.event.pull_request.base.repo.full_name == 'pypa/bandersnatch')
+      uses: codecov/codecov-action@v1
+      env:
+        OS: ${{ matrix.os }}
+        PYTHON: ${{ matrix.python-version }}
+      with:
+        env_vars: OS,PYTHON
+
+    - name: Run Integration Test
+      env:
+       TOXENV: INTEGRATION
+      run: |
+        python -m pip install . --use-feature=in-tree-build
+        python test_runner.py
+
+    - name: Run Integration Test (uvloop)
+      if: matrix.os != 'windows-latest'
+      env:
```

I feel this is left over? I don't think this can ever be hit now
```suggestion
    - name: Run Integration Test (uvloop)
      env:
```